docs/sw2sphinxopenapi.py

```diff
@@ -0,0 +1,141 @@
+"""Generate Markdown documents from an OpenAPI swagger file."""
+
+from __future__ import print_function
+
+import contextlib
+import os
+import os.path
+import re
+import sys
+import textwrap
+
+import yaml
+
+
+class OutputFiles(object):
+    """A context manager to manage a series of files.
+
+    Use like this::
+
+        with OutputFiles() as outfiles:
+            ...
+            if some_condition():
+                f = outfiles.open("filename.txt", "w")
+
+    Each open will close the previously opened file, and the end of the with
+    statement will close the last one.
```

Comment:
I didn't take the time to read the code of this context manager in details in your previous PR. I think this class is a little bit overkill and error-prone. For instance, take the following code:

```
f1 = outfiles.open(...)
f2 = outfiles.open(...)
# attempt (and fail) to do something with f1
```

My opinion is that the usual file context manager (`with open(...) as f:`) is easier to understand and refactor, and does not introduce much complexity.

Reply:
A classic with-statement won't work for the outfiles due to needing to write the first header at the top, and keep them open.  I've used explicit closes in the latest code.